src/mpid/ch4/netmod/ofi/ofi_init.h

```diff
@@ -1394,44 +1394,40 @@ static inline int MPIDI_OFI_application_hints(int rank)
     /* Seed the global settings values for cases where we are using runtime sets */
     MPIDI_Global.settings.enable_av_table =
         MPIR_CVAR_CH4_OFI_ENABLE_AV_TABLE !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_AV_TABLE : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_av_table :
-        MPIR_CVAR_CH4_OFI_ENABLE_AV_TABLE;
+        MPIDI_OFI_caps_list[prov_idx].enable_av_table : MPIR_CVAR_CH4_OFI_ENABLE_AV_TABLE;
     MPIDI_Global.settings.enable_scalable_endpoints =
         MPIR_CVAR_CH4_OFI_ENABLE_SCALABLE_ENDPOINTS !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_SCALABLE_ENDPOINTS : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_scalable_endpoints :
+        MPIDI_OFI_caps_list[prov_idx].enable_scalable_endpoints :
         MPIR_CVAR_CH4_OFI_ENABLE_SCALABLE_ENDPOINTS;
     /* If the user specifies -1 (=don't care) and the provider supports it, then try to use STX
      * and fall back if necessary in the RMA init code */
     MPIDI_Global.settings.enable_shared_contexts =
         MPIR_CVAR_CH4_OFI_ENABLE_SHARED_CONTEXTS !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_SHARED_CONTEXTS : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_shared_contexts :
+        MPIDI_OFI_caps_list[prov_idx].enable_shared_contexts :
         MPIR_CVAR_CH4_OFI_ENABLE_SHARED_CONTEXTS;
     MPIDI_Global.settings.enable_mr_scalable =
         MPIR_CVAR_CH4_OFI_ENABLE_MR_SCALABLE !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_MR_SCALABLE : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_mr_scalable :
-        MPIR_CVAR_CH4_OFI_ENABLE_MR_SCALABLE;
+        MPIDI_OFI_caps_list[prov_idx].enable_mr_scalable : MPIR_CVAR_CH4_OFI_ENABLE_MR_SCALABLE;
     MPIDI_Global.settings.enable_tagged =
         MPIR_CVAR_CH4_OFI_ENABLE_TAGGED !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_TAGGED : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_tagged :
-        MPIR_CVAR_CH4_OFI_ENABLE_TAGGED;
+        MPIDI_OFI_caps_list[prov_idx].enable_tagged : MPIR_CVAR_CH4_OFI_ENABLE_TAGGED;
     MPIDI_Global.settings.enable_am =
         MPIR_CVAR_CH4_OFI_ENABLE_AM !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_AM : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_am :
-        MPIR_CVAR_CH4_OFI_ENABLE_AM;
+        MPIDI_OFI_caps_list[prov_idx].enable_am : MPIR_CVAR_CH4_OFI_ENABLE_AM;
     MPIDI_Global.settings.enable_rma =
         MPIR_CVAR_CH4_OFI_ENABLE_RMA !=
         -1 ? MPIR_CVAR_CH4_OFI_ENABLE_RMA : prov_name ?
-        MPIDI_OFI_caps_list[MPIDI_OFI_get_set_number(prov_name)].enable_rma :
-        MPIR_CVAR_CH4_OFI_ENABLE_RMA;
+        MPIDI_OFI_caps_list[prov_idx].enable_rma : MPIR_CVAR_CH4_OFI_ENABLE_RMA;
     /* try to enable atomics only when RMA is enabled */
     MPIDI_Global.settings.enable_atomics =
         MPIDI_OFI_ENABLE_RMA ? (MPIR_CVAR_CH4_OFI_ENABLE_ATOMICS !=
```

Comment:
@alexander-sannikov you missed one on line 1439. Another thought: should we go further and just get a pointer to a `caps` struct (`MPIDI_OFI_capabilities_t`) rather than indexing into the array every time?

Reply:
@raffenet done. Could you please start testing.